File: Tema 05 - Typescript/after/PetShelterFrontend/src/models/Fundraiser.ts
import type { Person } from './Person';

export class Fundraiser {
  name: string;
  target: number;
  currentlyRaised?: number;
  dueDate: Date;
  creationDate?: Date;
  status?: FundraiserStatus;
  owner: Person;
  donors?: Person[];

  constructor(
    name: string,
    target: number,
    dueDate: Date,
    owner: Person,
    creationDate?: Date,
    status?: FundraiserStatus,
    currentlyRaised?: number,
    donors?: Person[],
  ) {
    this.name = name;
    this.target = target;
    this.currentlyRaised = currentlyRaised;
    this.creationDate = creationDate;
    this.dueDate = dueDate;
    this.status = status;
    this.owner = owner;
    this.donors = donors;
  }
}

export const enum FundraiserStatus {
  'Closed' = 0,
  'Active' = 1,
}

File: Tema 07 - Clean code/Before Tema 3/PetShelter.Api/Resources/Fundraiser.cs
namespace PetShelter.Api.Resources
{
    public class Fundraiser
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public decimal Target { get; set; }
        public decimal CurrentlyRaised { get; set; }
        public DateTime CreationDate { get; set; }
        public DateTime DueDate { get; set; }
        public FundraiserStatus Status { get; set; }
    }
}
